src/pages/MapViewer/components/Toolbox/components/item-assets/components/VisualiseAsset/TiTilerCustomisationPanel.tsx

```diff
@@ -0,0 +1,423 @@
+import { useEffect, useMemo, useState } from 'react';
+import './TiTilerCustomisationPanel.scss';
+
+import { GiHistogram } from 'react-icons/gi';
+import { TbMathMaxMin } from 'react-icons/tb';
+import { Tooltip } from 'react-tooltip';
+import {
+  Bar,
+  BarChart,
+  Tooltip as RechartsTooltip,
+  ResponsiveContainer,
+  XAxis,
+  YAxis,
+} from 'recharts';
+
+import { ClipboardButton } from '@/components/clipboard/ClipboardButton';
+import { StacAsset } from '@/typings/common';
+
+import {
+  availableColourMaps,
+  availableProjections,
+  determineAssetType,
+  isAssetKerchunk,
+} from './utils';
+
+type TiTilerCustomisationPanelProps = {
+  asset: StacAsset;
+};
+
+export const TiTilerCustomisationPanel = ({ asset }: TiTilerCustomisationPanelProps) => {
+  const assetType = useMemo(() => determineAssetType(asset), [asset]);
+
+  // URLs
+  const [generatedUrl, setGeneratedUrl] = useState('');
+  const [generatedPreviewUrl, setGeneratedPreviewUrl] = useState('');
+
+  // Common parameters
+  const [tileMatrixSetId, setTileMatrixSetId] = useState('WebMercatorQuad');
+  const [rescale, setRescale] = useState('');
+  const [colormapName, setColormapName] = useState('');
+
+  // Asset-specific
+  const isKerchunk = useMemo(() => isAssetKerchunk(asset), [asset]);
+  const [availableBands, setAvailableBands] = useState([]);
+  const [availableVariables, setAvailableVariables] = useState([]);
+  const [variable, setVariable] = useState('');
+  const [bidx, setBidx] = useState(1);
+
+  // Histogram data
+  const [histogramData, setHistogramData] = useState([]);
+
+  // Loading and error states
+  const [loadingVariables, setLoadingVariables] = useState(false);
+  const [variablesError, setVariablesError] = useState('');
+
+  const [loadingRescale, setLoadingRescale] = useState(false);
+  const [rescaleError, setRescaleError] = useState('');
+
+  const [loadingHistogram, setLoadingHistogram] = useState(false);
+  const [histogramError, setHistogramError] = useState('');
+
+  // Fetch available variables or bands
+  useEffect(() => {
+    const fetchData = async () => {
+      setLoadingVariables(true);
+      setVariablesError('');
+      try {
+        if (assetType === 'core') {
+          const url = `${import.meta.env.VITE_TITILER_CORE_ENDPOINT}/cog/info?url=${encodeURIComponent(
+            asset.href,
+          )}`;
+          const response = await fetch(url);
+          if (!response.ok) {
+            throw new Error('Failed to fetch band information.');
+          }
+          const data = await response.json();
```

Comment:
Do we want to have this fetch abstracted out as a method in case we want to make the request elsewhere in the code?

Reply:
Although these services won't be used elsewhere, I still agree that having fetching logic in a components code looks messy. 
I've moved as much as I can into `titilerServices.ts`